setup.py

```diff
@@ -1,15 +1,35 @@
 import setuptools
 from os import path
-from distutils.command.build_ext import build_ext as DistUtilsBuildExt
-from setuptools import setup, find_packages
+from setuptools import setup, find_packages, Extension
+import numpy as np
+from Cython.Build import cythonize
+
+CYTHON_DEBUG = False
+
+if CYTHON_DEBUG:
+    from Cython.Compiler.Options import get_directive_defaults
+    directive_defaults = get_directive_defaults()
+
+    directive_defaults['linetrace'] = True
+    directive_defaults['binding'] = True
+
+CYTHON_MACROS = [('CYTHON_TRACE', '1')] if CYTHON_DEBUG else None
 
 VERSION = '0.2.9'
 
+with open(path.join(path.dirname(__file__), 'requirements.txt')) as req_file:
+    requirements = req_file.read().splitlines()
```

Comment:
Heads up, I tried this before with `setup.py` and it failed to deploy to PyPI correctly. If it fails again, a quick fix would be to revert to the whole "list out everything" method.